agent/src/test/java/com/newrelic/agent/android/util/OfflineStorageTest.java

```diff
@@ -0,0 +1,145 @@
+        for (Map.Entry<String, String> entry : testKeyValue.entrySet()) {
+            String key = entry.getKey();
+            String value = entry.getValue();
+            Assert.assertEquals(instance.getOfflineFilePath(), key);
+            Assert.assertEquals("{'testKey': 'testValue'}", value);
+        }
+
+        double totalSize = instance.getTotalFileSize();
+        Assert.assertNotEquals(0, totalSize);
+    }
+
+    @Test
+    public void testGetAllOfflineData() {
+        try {
+            instance = new OfflineStorage(spyContext);
+            instance.cleanOfflineFiles();
+
+            Map<String, String> dataSet1 = instance.getAllOfflineData();
+            Assert.assertEquals(0, dataSet1.size());
+
+            instance.persistDataToDisk("{'testKey': 'testValue'}");
+            Thread.sleep(1000);
+            instance.persistDataToDisk("{'testKey': 'testValue'}");
+            Thread.sleep(1000);
+            instance.persistDataToDisk("{'testKey': 'testValue'}");
+            Thread.sleep(1000);
+            instance.persistDataToDisk("{'testKey': 'testValue'}");
+            Thread.sleep(1000);
+            instance.persistDataToDisk("{'testKey': 'testValue'}");
+            Thread.sleep(1000);
+
+            Map<String, String> dataSet2 = instance.getAllOfflineData();
+            Assert.assertEquals(5, dataSet2.size());
+
+            instance.cleanOfflineFiles();
+            Map<String, String> dataSet3 = instance.getAllOfflineData();
+            Assert.assertEquals(0, dataSet3.size());
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Test
+    public void testGetTotalFileSize() {
+        instance = new OfflineStorage(spyContext);
+        instance.cleanOfflineFiles();
+        double totalSize = instance.getTotalFileSize();
+        Assert.assertEquals(0, totalSize, 0);
+
+        instance.persistDataToDisk("{'testKey': 'testValue'}");
+
+        double totalSize2 = instance.getTotalFileSize();
+        Assert.assertNotEquals(0, totalSize2);
+    }
+
+    @Test
+    public void testDefaultMaxSize() {
+        instance = new OfflineStorage(spyContext);
+        instance.cleanOfflineFiles();
+
+        int defaultSize1 = instance.getOfflineStorageSize();
+        Assert.assertEquals(100 * 1024 * 1024, defaultSize1);
+
+        OfflineStorage.setMaxOfflineStorageSize(-1);
+        int defaultSize2 = instance.getOfflineStorageSize();
+        Assert.assertEquals(100 * 1024 * 1024, defaultSize2);
+
+        OfflineStorage.setMaxOfflineStorageSize(10);
+        int defaultSize3 = instance.getOfflineStorageSize();
+        Assert.assertEquals(10, defaultSize3);
+
+        instance.setOfflineStorageSize(-1);
+        int defaultSize4 = instance.getOfflineStorageSize();
+        Assert.assertEquals(-1, defaultSize4);
+
+        instance.setOfflineStorageSize(10);
+        int defaultSize5 = instance.getOfflineStorageSize();
+        Assert.assertEquals(10, defaultSize5);
+    }
+}
```

Comment:
Maybe add tests for concurrency (is it thread safe?), and UI blocking (must ever be called on main thread).